src/java/fr/paris/lutece/plugins/appointment/service/AppointmentService.java

```diff
@@ -291,6 +291,49 @@ public static List<Appointment> findListAppointmentsByFilter( AppointmentFilterD
         return AppointmentHome.findByFilter( appointmentFilter );
     }
 
+    /**
+     * Find a list of appointments ids matching the filter
+     *
+     * @param appointmentFilter
+     *         the filter
+     * @return a list of appointments
+     */
+    public static List<Integer> findListAppointmentsIdsByFilter( AppointmentFilterDTO appointmentFilter )
+    {
+        return AppointmentHome.findIdsByFilter( appointmentFilter );
+    }
+
+    public static List<AppointmentDTO> findListAppointmentsDTOByFilterByPage( AppointmentFilterDTO appointmentFilter, List<Integer> listIds, int nItemsPerPage,
```

Comment:
Could you please move this method, either as a private method within the AppointmentJspBean class or into a utility class

Reply:
the remark concerns the entire findListAppointmentsDTOByFilterByPage method or only the piece of code which calculates the pagination of the IDS?